### DISCUSSION TITLE
sha256 of a git repository

### DISCUSSION THREAD

**Discussion by alexbodn** [termux/termux-packages#20908] [Q&A] [answered]
hello friends,
i wish to build a package based, in the beginning, on the content of a git repository.
this seems to work, with TERMUX_PKG_SRCURL set to my git url.
however, TERMUX_PKG_SHA256 is still requested, and i don't know how to calculate it, since a repo is not a file.
is there any means to bypass the sha256 requirement for repositories?

**Comment by TomJo2000** [comment, marked as answer]
#### If the `TERMUX_PKG_SRCURL` is a git URL `TERMUX_PKG_SHA256` is not used.
https://github.com/termux/termux-packages/blob/75850a446708376ca7d1dad558c720ec99138d6c/scripts/build/get_source/termux_step_get_source.sh#L4-L5
Git source URLs should have a `git+` prefix to be identified as such.
See LuaJIT as an example.
https://github.com/termux/termux-packages/blob/75850a446708376ca7d1dad558c720ec99138d6c/packages/libluajit/build.sh#L6

---

`termux_git_clone_src` is implemented here.
https://github.com/termux/termux-packages/blob/75850a446708376ca7d1dad558c720ec99138d6c/scripts/build/get_source/termux_git_clone_src.sh#L1-L45
<sup>(This function could probably use a rewrite, but I just rewrote the license installation step and I don't feel like breaking prod twice in one week.)</sup>
